### PR TITLE
gsdx-d3d11: Enable HLSL debugging on development builds

### PR DESCRIPTION
**Pull request by tadanokojin**
Enables live debugging for D3D11 shader code.

### CODE REVIEW

**Comment by gregory38** [comment]
Maybe create a get_compiler_flags ?

**Comment by gregory38** [comment]
Yes it can surely be better. Maybe a void* can help too

File: plugins/GSdx/Renderers/DX11/GSDevice11.h [hunk 1]
```diff
@@ -232,9 +232,11 @@ class GSDevice11 : public GSDeviceDX
...
-	void CompileShader(const char* source, size_t size, const char* fn, ID3DInclude *include, const char* entry, D3D_SHADER_MACRO* macro, ID3D11GeometryShader** gs, D3D11_SO_DECLARATION_ENTRY* layout, int count);
-	void CompileShader(const char* source, size_t size, const char* fn, ID3DInclude *include, const char* entry, D3D_SHADER_MACRO* macro, ID3D11PixelShader** ps);
+	void CreateShader(const char* source, size_t size, const char* fn, ID3DInclude *include, const char* entry, D3D_SHADER_MACRO* macro, ID3D11VertexShader** vs, D3D11_INPUT_ELEMENT_DESC* layout, int count, ID3D11InputLayout** il);
+	void CreateShader(const char* source, size_t size, const char* fn, ID3DInclude *include, const char* entry, D3D_SHADER_MACRO* macro, ID3D11GeometryShader** gs);
+	void CreateShader(const char* source, size_t size, const char* fn, ID3DInclude *include, const char* entry, D3D_SHADER_MACRO* macro, ID3D11GeometryShader** gs, D3D11_SO_DECLARATION_ENTRY* layout, int count);
+	void CreateShader(const char* source, size_t size, const char* fn, ID3DInclude *include, const char* entry, D3D_SHADER_MACRO* macro, ID3D11PixelShader** ps);
+
```
Comment (lightningterror) [inline plugins/GSdx/Renderers/DX11/GSDevice11.h]: Empty row or intended ?

Comment (tadanokojin) [reply plugins/GSdx/Renderers/DX11/GSDevice11.h]: Intended to separate the CreateShaders from the CompileShader.

**Comment by tadanokojin** [comment]
@gregory38 how's this for a solution? I abstracted the compile code from the functions so it's only used in one place.

**Comment by gregory38** [comment]
Do we use geometry shader with stream ?

It is better. Note on ogl. You have shader which are compiled and then linked to a program (or ProgramShader).

**Comment by tadanokojin** [comment]
> Do we use geometry shader with stream ?

I'm no shader guy but do you mean like this?
https://github.com/PCSX2/pcsx2/blob/4a0697fda255fe3ed9643082726f23cc283539c3/plugins/GSdx/res/tfx.fx#L716

If so, yes.

**Pull request merged**